src/components/application_manager/src/hmi_interfaces_impl.cc

```diff
@@ -0,0 +1,220 @@
+  convert_map[Navigation_OnVideoDataStreaming] =
+      HmiInterfaces::HMI_INTERFACE_Navigation;
+  convert_map[Navigation_GetWayPoints] =
+      HmiInterfaces::HMI_INTERFACE_Navigation;
+  convert_map[Navigation_OnWayPointChange] =
+      HmiInterfaces::HMI_INTERFACE_Navigation;
+  convert_map[Navigation_SubscribeWayPoints] =
+      HmiInterfaces::HMI_INTERFACE_Navigation;
+  convert_map[Navigation_UnsubscribeWayPoints] =
+      HmiInterfaces::HMI_INTERFACE_Navigation;
+  convert_map[VehicleInfo_IsReady] = HmiInterfaces::HMI_INTERFACE_VehicleInfo;
+  convert_map[VehicleInfo_GetVehicleType] =
+      HmiInterfaces::HMI_INTERFACE_VehicleInfo;
+  convert_map[VehicleInfo_ReadDID] = HmiInterfaces::HMI_INTERFACE_VehicleInfo;
+  convert_map[VehicleInfo_GetDTCs] = HmiInterfaces::HMI_INTERFACE_VehicleInfo;
+  convert_map[VehicleInfo_DiagnosticMessage] =
+      HmiInterfaces::HMI_INTERFACE_VehicleInfo;
+  convert_map[VehicleInfo_SubscribeVehicleData] =
+      HmiInterfaces::HMI_INTERFACE_VehicleInfo;
+  convert_map[VehicleInfo_UnsubscribeVehicleData] =
+      HmiInterfaces::HMI_INTERFACE_VehicleInfo;
+  convert_map[VehicleInfo_GetVehicleData] =
+      HmiInterfaces::HMI_INTERFACE_VehicleInfo;
+  convert_map[VehicleInfo_OnVehicleData] =
+      HmiInterfaces::HMI_INTERFACE_VehicleInfo;
+  convert_map[SDL_ActivateApp] = HmiInterfaces::HMI_INTERFACE_SDL;
+  convert_map[SDL_GetUserFriendlyMessage] = HmiInterfaces::HMI_INTERFACE_SDL;
+  convert_map[SDL_OnAllowSDLFunctionality] = HmiInterfaces::HMI_INTERFACE_SDL;
+  convert_map[SDL_OnReceivedPolicyUpdate] = HmiInterfaces::HMI_INTERFACE_SDL;
+  convert_map[SDL_OnPolicyUpdate] = HmiInterfaces::HMI_INTERFACE_SDL;
+  convert_map[SDL_GetListOfPermissions] = HmiInterfaces::HMI_INTERFACE_SDL;
+  convert_map[SDL_OnAppPermissionConsent] = HmiInterfaces::HMI_INTERFACE_SDL;
+  convert_map[SDL_OnAppPermissionChanged] = HmiInterfaces::HMI_INTERFACE_SDL;
+  convert_map[SDL_OnSDLConsentNeeded] = HmiInterfaces::HMI_INTERFACE_SDL;
+  convert_map[SDL_UpdateSDL] = HmiInterfaces::HMI_INTERFACE_SDL;
+  convert_map[SDL_GetStatusUpdate] = HmiInterfaces::HMI_INTERFACE_SDL;
+  convert_map[SDL_OnStatusUpdate] = HmiInterfaces::HMI_INTERFACE_SDL;
+  convert_map[SDL_OnSystemError] = HmiInterfaces::HMI_INTERFACE_SDL;
+  convert_map[SDL_AddStatisticsInfo] = HmiInterfaces::HMI_INTERFACE_SDL;
+  convert_map[SDL_GetURLS] = HmiInterfaces::HMI_INTERFACE_SDL;
+  return convert_map;
+}
+
+HmiInterfaces::InterfaceState HmiInterfacesImpl::GetInterfaceState(
+    HmiInterfaces::InterfaceName interface) const {
+  const InterfaceStatesMap::const_iterator it =
+      interfaces_states_.find(interface);
+  // all interfaces should be presented in interfaces_states_ map.
+  DCHECK(it != interfaces_states_.end());
+  return it->second;
+}
+
+void HmiInterfacesImpl::SetIntefaceState(HmiInterfaces::InterfaceName interface,
+                                         HmiInterfaces::InterfaceState state) {
+  interfaces_states_[interface] = state;
```

Comment:
@LuxoftAKutsan,
Also need to `DCHECK(interfaces_states_.find(interface))`
BTW, where is setting all interfaces to initial state?